File: app/src/main/java/com/ninhttd/devtest/data/local/dao/ProductDao.java
package com.ninhttd.devtest.data.local.dao;

import androidx.room.Dao;
import androidx.room.Query;

import com.ninhttd.devtest.base.BaseDao;
import com.ninhttd.devtest.data.entity.ProductEntity;

import java.util.List;

import io.reactivex.Single;

@Dao
public interface ProductDao extends BaseDao<ProductEntity> {
    @Query("select * from ProductEntity")
    Single<List<ProductEntity>> getAll();

    @Query("select * from ProductEntity")
    List<ProductEntity> getAllNoThread();

    @Query("select * from ProductEntity where mDisplayName like :key")
    Single<List<ProductEntity>> search(String key);
}

File: app/src/main/java/com/ninhttd/devtest/data/local/dao/ProductDao.kt
package com.ninhttd.devtest.data.local.dao

import androidx.room.Dao
import androidx.room.Query

import com.ninhttd.devtest.base.BaseDao
import com.ninhttd.devtest.data.entity.ProductEntity

import io.reactivex.Single

@Dao
interface ProductDao : BaseDao<ProductEntity> {
    @get:Query("select * from ProductEntity")
    val all: Single<List<ProductEntity>>

    @get:Query("select * from ProductEntity")
    val allNoThread: List<ProductEntity>

    @Query("select * from ProductEntity where displayName like :key")
    fun search(key: String): Single<List<ProductEntity>>
}
